table/namespace_classifier.go

```diff
@@ -142,14 +143,15 @@ func (c *tableNamespaceClassifier) GetRegionNamespace(regionInfo *core.RegionInf
 	c.RLock()
 	defer c.RUnlock()
 
+	isMeta := Key(regionInfo.StartKey).IsMeta()
 	tableID := Key(regionInfo.StartKey).TableID()
-	if tableID == 0 {
+	if tableID == 0 && !isMeta {
 		return namespace.DefaultNamespace
 	}
 
 	for name, ns := range c.nsInfo.namespaces {
 		_, ok := ns.TableIDs[tableID]
-		if ok {
+		if ok || (isMeta && ns.Meta) {
 			return name
```

Comment:
any test to cover meta here?

Reply:
yes. See https://github.com/pingcap/pd/pull/834/files#diff-adf781e5159cb131c61ba6e5f5e4778eR110